service/handlers.go

```diff
@@ -6,25 +6,21 @@ import (
 	"github.com/microdevs/missy/log"
 )
 
-func StartTimerHandler(h Handler) Handler {
-	return HandlerFunc(func(w *ResponseWriter, r *http.Request) {
-		context.Set(r, RequestTimer, NewTimer())
-		h.ServeHTTP(w,r)
-	})
-}
 
-func AccessLogHandler(h Handler) Handler {
-	return HandlerFunc(func(w *ResponseWriter, r *http.Request) {
-		log.Infof("%s \"%s %s %s\" %d - %s", r.RemoteAddr, r.Method, r.URL, r.Proto, w.Status, r.UserAgent())
-		h.ServeHTTP(w,r)
-	})
+func AccessLogHandler(h http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		rec := ResponseWriter{w, http.StatusOK}
+		h(&rec,r)
+		log.Infof("%s \"%s %s %s\" %d - %s", r.RemoteAddr, r.Method, r.URL, r.Proto, rec.Status, r.UserAgent())
+	}
 }
 
-func StopTimerHandler(h Handler) Handler {
-	return HandlerFunc(func(w *ResponseWriter, r *http.Request) {
-		h.ServeHTTP(w,r)
-		timer := context.Get(r, RequestTimer).(*Timer)
+func MeasureTimeHandler(h http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		t := NewTimer()
+		rec := ResponseWriter{w, http.StatusOK}
```

Comment:
What about no StatusOK from response?

Reply:
It's not hardcoded, just the initial value. Have a look at `service.go` where I do a `w.WriteHeader(code)`.

Having said that: There should be a test that verifies that the code is actually updated.

Reply:
why setting a value there at all? It's really confusing.

Reply:
@levrik  Look at the standard library. Here's what's httptest is using:
```
func NewRecorder() *ResponseRecorder {
	return &ResponseRecorder{
		HeaderMap: make(http.Header),
		Body:      new(bytes.Buffer),
		Code:      200,
	}
}
```

I am not a Gopher yet, but from my limited understanding Go likes to set sane default values. I think 200 is a sane default value, but I am totally open for your suggestions and experiences with Go.

Reply:
@learnjin mhhh I see your point. But your example creates a new struct instance with defaults. I think our usecase differs a bit. Let's talk about that on Monday 🙂